### PR TITLE
feat: first execution context skeleton

### PR DESCRIPTION
**Pull request by enitrat**
Skeleton of a first draft of the ExecutionContext.
This PR doesn't look into any optimisation and just provides an implementation with the bare minimum so that we can continue working towards the first opcode. Once we will have a clearer view of the requirements after trying multiple things with Cairo 1, we will think about the refactoring and have a final version of it.
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #43 

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/tests/test_kakarot.cairo [hunk 1]
```diff
@@ -8,14 +8,16 @@ use kakarot::context::CallContext;
 #[test]
 #[available_gas(2000000)]
 #[should_panic(expected: (0, ))]
+#[ignore]
```
Comment (danilowhk) [inline src/tests/test_kakarot.cairo:11]: What is the current issue with this test?

File: src/evm.cairo [hunk 2]
```diff
@@ -5,15 +5,15 @@ use kakarot::context::ExecutionContextTrait;
...
-    // Create new execution context.
-    let mut ctx = ExecutionContextTrait::new(call_context);
-    // Compute the intrinsic gas cost for the current transaction and increase the gas used.
-    ctx.process_intrinsic_gas_cost();
-    // Print the execution context.
-    ctx.print_debug();
-    let mut evm_instructions = EVMInstructionsTrait::new();
-    // Execute the transaction.
-    evm_instructions.run(ref ctx)
+fn execute(call_context: CallContext) { /// TODO
```
Comment (danilowhk) [inline src/evm.cairo:8]: Could we detail what is the TODO?

File: src/tests/test_execution_context.cairo [hunk 3]
```diff
@@ -0,0 +1,4 @@
+use kakarot::context::{CallContextTrait, ExecutionContextTrait};
+// TODO
```
Comment (danilowhk) [inline src/tests/test_execution_context.cairo:2]: Any specific reason why not add some tests now?

Comment (enitrat) [reply src/tests/test_execution_context.cairo:2]: Not really, I thought about doing them in another PR, if you think I should always do it in the same one I will do so :)

**Comment by enitrat** [comment]
@danilowhk I added tests for what's currently implemented

**Review by danilowhk — APPROVED** [review]

**Pull request merged**